### ISSUE TITLE
Zero Exponent in unyt

### ISSUE THREAD

**Issue by JBorrow** [yt-project/unyt#203]
* unyt version: unyt v2.8.0
* Python version: Python 3.9.7
* Operating System: MacOS

### Description

Taking unyt_quantity to zero exponent returns original units and not (dimensionless).

Is this expected behaviour?

### What I Did

```
import platform
import unyt

print("Python", platform.python_version())
print("unyt", unyt.__version__)

base_quantity = unyt.unyt_quantity(112.0, "g")

print("Raw", base_quantity, "expected g")
print("Small Exponent", base_quantity**0.03, "expected g**0.03",)
print("Zero Exponent (float)", base_quantity**(0.0), "expected (dimensionless)")
print("Zero Exponent (int)", base_quantity**0, "expected (dimensionless)")
```

Output:
```
Python 3.9.7
unyt v2.8.0
Raw 112.0 g expected g
Small Exponent 1.1520638270207721 g**(3/100) expected g**0.03
Zero Exponent (float) 1.0 g expected (dimensionless)
Zero Exponent (int) 1.0 g expected (dimensionless)
```

**Comment by ngoldbaum** [comment]
I think you’ll need to edit the `__pow__` implementation for the `Unit` class: https://github.com/yt-project/unyt/blob/e86c0500e9c9f29914eed3ef7d8f11c98768eedd/unyt/unit_object.py#L469

You may also need to touch this code path in the `unyt_array` class: https://github.com/yt-project/unyt/blob/e86c0500e9c9f29914eed3ef7d8f11c98768eedd/unyt/array.py#L1758

I definitely agree this is a bug. I don’t have time right now to work on a fix.

**Comment by JBorrow** [comment]
Looks like when using a power of zero, we default to the ufunc `_ones_like`. I'm guessing this is a numpy optimisation, and we'll need to just make sure that the right units (dimensionless) gets associated in this case.

```
In [13]: x = unyt.unyt_quantity(4.0, "g")

In [14]: pdb.run('x**0.0')
> <string>(1)<module>()
(Pdb) s
--Call--
> /Users/jborrow/Documents/OpenSource/unyt/unyt/array.py(1694)__array_ufunc__()
-> def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
(Pdb) ufunc
<ufunc '_ones_like'>
```

**Comment by ngoldbaum** [comment]
Ah, fun! That might make things harder if there’s
no way to determine if the original ufunc was a pow.

**Comment by JBorrow** [comment]
Closed by #204
